src/parse.rs

```diff
@@ -79,16 +91,20 @@ impl<'a> Parser<'a> {
         }
     }
 
-    pub fn array_step(&mut self) -> JsonResult<bool> {
+    pub fn array_step(&mut self) -> JsonResult<Option<Peak>> {
```

Comment:
I didn't have a chance to look at performance implications of this change, it seemed necessary in order to support trailing comma as an error.

Reply:
it's good.